### PR TITLE
PLT-6713: add /search command

### PR DESCRIPTION
**Pull request by ccbrown**
#### Summary
/search [text] slash command: Search text in messages

[Demo](https://gfycat.com/LeanCourageousImperatorangel)

#### Ticket Link
https://github.com/mattermost/platform/issues/6556

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates

### CODE REVIEW

**Comment by ccbrown** [comment]
@esethna That's a pre-existing bug, not specific to search. The error occurs when you do _anything_ that opens up the sidebar. For example, just click the flag button in the top right and you'll get the same error.

On master, that bug is currently fixed, but another aspect of search is broken (on pre-release right now, many searches completely fail with status 400 errors).

File: webapp/actions/channel_actions.jsx [hunk 1]
```diff
@@ -76,6 +76,9 @@ export function executeCommand(message, args, success, error) {
     msg = cmd + msg.substring(cmdLength, msg.length);
 
     switch (cmd) {
+    case '/search':
+        GlobalActions.emitSearchEvent(msg.substring(cmdLength + 1, msg.length));
```
Comment (hmhealey) [inline webapp/actions/channel_actions.jsx]: We actually already have a function for this in `post_actions.jsx`. Could you switch this to use that one?

Comment (ccbrown) [inline webapp/actions/channel_actions.jsx]: Done.

**Comment by hmhealey** [comment]
Looks good. You've got a merge conflict now, so you'll need to rebase, but I'd wait for Corey's review so you don't end up needing to do it twice

**Pull request merged**